### PR TITLE
feature(deploy): add AWS deploy Github action

### PR DESCRIPTION
**Pull request by davidholyko**

### CODE REVIEW

File: .github/workflows/aws.yaml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        id: branch-name
+        run: |
+          echo ::set-output name=SOURCE_NAME::${GITHUB_REF#refs/*/}
+          echo ::set-output name=SOURCE_BRANCH::${GITHUB_REF#refs/heads/}
+          echo ::set-output name=SOURCE_TAG::${GITHUB_REF#refs/tags/}
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
```
Comment (davidholyko) [inline .github/workflows/aws.yaml:54]: looks like this repo needs secret keys. for 

```
AWS_ACCESS_KEY_ID
AWS_SECRET_ACCESS_KEY
```

File: .github/workflows/aws.yaml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-1
+
+      - name: Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+
+      - name: Build Backend Code, tag, and push image to Amazon ECR
```
Comment (davidholyko) [inline .github/workflows/aws.yaml:62]: it seems this step should be replaced.

File: .github/workflows/aws.yaml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+          ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+          ECR_REPOSITORY: hrbot
+          IMAGE_TAG: ${{ github.sha }}
+        run: |
+          # Build a docker container and push it to ECR so that it can be deployed to ECS.
+          docker build -f api/Dockerfile -t $ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG api/
+          docker push $ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG
+          echo "::set-output name=image::$ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG"
+
+      - name: Fill in the new image ID in the Amazon ECS task definition
```
Comment (davidholyko) [inline .github/workflows/aws.yaml:74]: it seems this step should be replaced.

File: .github/workflows/aws.yaml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        id: fill-in-php-docker
+        uses: aws-actions/amazon-ecs-render-task-definition@v1
+        with:
+          task-definition: api/task-definition.json
+          container-name: scrum-backend
+          image: ${{ steps.build-php-image.outputs.image }}
+
+      - name: Deploy Amazon ECS task definition
```
Comment (davidholyko) [inline .github/workflows/aws.yaml:82]: not sure if we have any php stuff

File: .github/workflows/aws.yaml (hunk 1 above)
Comment (fabioelia) [reply .github/workflows/aws.yaml:54]: I can give this repo access 👍

File: .github/workflows/aws.yaml (hunk 2 above)
Comment (fabioelia) [reply .github/workflows/aws.yaml:62]: You can rename it, but this is important to build the docker container. 
This setup might be a bit more relevant for you https://github.com/f1v/camino-for-good-app

File: .github/workflows/aws.yaml (hunk 3 above)
Comment (fabioelia) [reply .github/workflows/aws.yaml:74]: needed

File: .github/workflows/aws.yaml (hunk 4 above)
Comment (fabioelia) [reply .github/workflows/aws.yaml:82]: needed :)

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,6 @@
+FROM node:14-slim
+WORKDIR /app
+COPY package.json /app
+RUN npm install
```
Comment (fabioelia) [inline Dockerfile:4]: do a CI install

File: aws/task-definition.json [hunk 6]
```diff
@@ -0,0 +1,140 @@
+{
```
Comment (fabioelia) [inline aws/task-definition.json:1]: does your container need anything like a database or relying on other servers?

Comment (davidholyko) [reply aws/task-definition.json:1]: currently doessn't rely on database. Later functionality could use database to keep track fo paymo information and hide it from other users

File: aws/task-definition.json [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+          "protocol": "tcp",
+          "containerPort": 80
+        }
+      ],
+      "command": null,
+      "linuxParameters": null,
+      "cpu": 0,
+      "environment": [
+        {
+          "name": "APP_ENV",
```
Comment (davidholyko) [inline aws/task-definition.json:22]: prod hrbot will need its own ENV variables

File: aws/task-definition.json [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+          "value": "prod"
+        },
+        {
+          "name": "DATABASE_URL",
```
Comment (davidholyko) [inline aws/task-definition.json:26]: hrbot currently has no db

File: aws/task-definition.json [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+      "dnsServers": null,
+      "mountPoints": [],
+      "workingDirectory": null,
+      "secrets": null,
+      "dockerSecurityOptions": null,
+      "memory": null,
+      "memoryReservation": 256,
+      "volumesFrom": [],
+      "stopTimeout": null,
+      "image": "538980331939.dkr.ecr.us-east-1.amazonaws.com/bulego-api:88c6a93478cadfa5675f27162a35ddd63e655797",
```
Comment (davidholyko) [inline aws/task-definition.json:41]: probably dont want to point image to bulego :D

File: aws/task-definition.json [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+      "hostname": null,
+      "extraHosts": null,
+      "pseudoTerminal": null,
+      "user": null,
+      "readonlyRootFilesystem": null,
+      "dockerLabels": null,
+      "systemControls": null,
+      "privileged": null,
+      "name": "hrbot-app"
+    },
```
Comment (davidholyko) [inline aws/task-definition.json:59]: end of first container definition. Not sure if we need a second

File: package.json [hunk 11]
```diff
@@ -6,7 +6,9 @@
   "scripts": {
     "start": "nodemon index.js",
     "semantic-release": "semantic-release",
-    "test": "echo 'No Testing Done ;)'"
+    "test": "echo 'No Testing Done ;)'",
+    "docker:build": "docker build -t hrbot-app .",
+    "docker:start": "docker run -p 3535:3535 -it hrbot-app"
```
Comment (davidholyko) [inline package.json:11]: used locally, not sure how prod will use these

File: .github/workflows/aws.yaml [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+        uses: aws-actions/amazon-ecr-login@v1
+
+      - name: Build Backend Code, tag, and push image to Amazon ECR
+        id: build-php-image
+        env:
+          ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+          ECR_REPOSITORY: hrbot
+          IMAGE_TAG: ${{ github.sha }}
+        run: |
+          # Build a docker container and push it to ECR so that it can be deployed to ECS.
```
Comment (davidholyko) [inline .github/workflows/aws.yaml:69]: steps might be different for hrbot. we dont have `api` folder

File: src/app.js [hunk 13]
```diff
@@ -131,4 +132,9 @@ app.event('app_mention', async ({ event, say }) => {
   }
 });
 
+receiver.router.get('/health', (req, res) => {
+  // You're working with an express req and res now.
+  res.send('david is happy!');
```
Comment (davidholyko) [inline src/app.js:137]: :smile:

File: .github/workflows/aws.yaml [hunk 14]
```diff
@@ -0,0 +1,92 @@
...
+#    See the documentation for each action used below for the recommended IAM policies for this IAM user,
+#    and best practices on handling the access key credentials.
+
+# on:
+#   push:
+#     branches:
+#       - master
+#   release:
+#     types: [created]
+on: [push]
```
Comment (davidholyko) [inline .github/workflows/aws.yaml:32]: does this mean that a push to any branch, it runs this deploy action?

Comment (fabioelia) [reply .github/workflows/aws.yaml:32]: I leave this up to you when you want it deployed.

This isn't like netlify, so it won't build a new temporary env per build / pr.
This is meant to be your prod release, use it wisely :)

File: package.json (hunk 11 above)
Comment (fabioelia) [reply package.json:11]: not really relevant for the builds, you can take a look at the AWS files if you want to extract it here.

**Pull request merged**